lib/octopoller.rb

```diff
@@ -11,24 +11,18 @@ class TimeoutError < StandardError; end
   # error_handler - A proc that will run with each instance of an error
   # yield - A block that will execute, and if it raises an error it will re-run until success or the timeout is reached
   # raise - Raises an Octopoller::TimeoutError if the timeout is reached
-  #
-  # rubocop:disable MethodLength
-  def poll(wait: 1, timeout: 15, error_handler: nil)
+  def poll(wait: 1, timeout: 15)
```

Comment:
This can happen in another PR or really it doesn't have to happen at all, but I think some polling use cases would rather use a `max_retries` option instead of a `timeout` in seconds argument. E.g. "I want to poll a maximum of 5 times". In our case, seconds works fine, but I could see octokit wanting to use a maximum retries system. That's how some of their polling logic is currently implemented.

Reply:
I like that idea. Also a polling with exponential backoff would be nice. I'll definitely make that in a follow up PR.